addons/pos_viva_wallet/static/src/app/payment_viva_wallet.js

```diff
@@ -0,0 +1,199 @@
+/** @odoo-module */
+
+import { _t } from "@web/core/l10n/translation";
+import { PaymentInterface } from "@point_of_sale/app/payment/payment_interface";
+import { ErrorPopup } from "@point_of_sale/app/errors/popups/error_popup";
+import { sprintf } from "@web/core/utils/strings";
+import { uuidv4 } from "@point_of_sale/utils";
+
+export class PaymentVivaWallet extends PaymentInterface {
+
+    /*
+     Developer documentation:
+    https://developer.vivawallet.com/apis-for-point-of-sale/card-terminals-devices/rest-api/eft-pos-api-documentation/
+    */
+
+    setup() {
+        super.setup(...arguments);
+        this.paymentLineResolvers = {};
+    }
+    send_payment_request(cid) {
+        super.send_payment_request(cid);
+        return this._viva_wallet_pay(cid);
+    }
+    send_payment_cancel(order, cid) {
+        super.send_payment_cancel(order, cid);
+        return this._viva_wallet_cancel();
+    }
+    set_most_recent_service_id(id) {
+        this.most_recent_service_id = id;
+    }
```

Comment:
`most_recent_service_id` seems to be unused here in viva_wallet implementation. Maybe you can remove calls to `set_most_recent_service_id`.